enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,504 @@
+  definition of the cluster I want to deploy as input to the tool.
+- As an OpenShift Infrastructure owner, I must be able to retrieve the
+  validation errors from an in-progress or failed deployment in a
+  programmatic way.
+- As an OpenShift Infrastructure owner, I must be able to get the
+  events and progress of the installation in a programmatic way.
+- As an OpenShift Infrastructure owner, I must be able to retrieve the
+  kubeconfig and OpenShift Console URL for my new cluster in a
+  programmatic way.
+
+### API Extensions
+
+This enhancement describes changes to the installer, but will not
+introduce any in-cluster API changes via CRDs, webhooks, or aggregated
+API servers.
+
+### Implementation Details/Notes/Constraints [optional]
+
+We already have several ways to describe OpenShift clusters for our
+different installation workflows, and want to avoid inventing a new
+one. Therefore we will reuse existing inputs relevant to on-premises
+deployment, borrowing from the IPI installer and other tools such as
+the zero-touch provisioning (ZTP) system in multi-cluster engine
+(MCE).
+
+Users often do not want to store the credentials for the baseboard
+management controllers (BMC) of their hosts inside of the cluster they
+are deploying. Therefore we cannot rely on them to give us those
+credentials during installation, and we must support tools outside of
+OpenShift managing the hosts' boot process. This places more of the
+burden of designing the end-to-end automated workflow on users or our
+solution architects.
+
+### Risks and Mitigations
+
+The inputs to building an on-premises cluster, especially the per-host
+static network settings, can be complex. Manually creating the inputs
+to the installer provides many opportunities to introduce errors. For
+the automated use case, we assume that users are likely to mitigate
+this risk by also automating production of the input files using
+templates or file generators. Other mitigation approaches, such as a
+GUI for constructing the files, are left to future enhancements.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. What do we do with the installation logs?
+
+   - Write them to disk on each host in a way that allows someone
+     logging in to find them or allows a Pod to mount the directory
+     they are in and `cat` them to be collected via the OpenShift API.
+   - Expect the orchestration tool to download them before allowing
+     deployment to continue.
+   - Can we take advantage of or replicate the installer's [bootstrap
+     service
+     records](https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/docs/dev/bootstrap_services.md)
+     behavior?
+
+2. Is the workflow different for single-node? Do we need to start the
+   service and go through the orchestration steps for one node? If we
+   don't, how do we handle the validations?
+3. How do we implement a feature flag in the installer?
```

Comment:
We discussed using a build flag, which is something we already do for ARO & OKD.